File: Week 2/id_545/LeetCode_49_545.py
## 49. 字母异位词分组

# 哈希
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dct = collections.defaultdict(list)
        for word in strs:
            dct[tuple(sorted(word))].append(word)
        return dct.values()

File: Week 2/id_545/LeetCode_49_545.js
// 49. 字母异位词分组

// 哈希
var groupAnagrams = function(strs) {
    let arr = []
    let dct = {}
    for (let i in strs) {
        let tmp = strs[i].split('').sort().join('')
        tmp in dct ? dct[ tmp ].push(strs[i]) : dct[ tmp ] = [strs[i]]
    }
    for (let i in dct) {
        arr.push(dct[i])
    }
    return arr
};

